### ISSUE TITLE
postfix plugin not working

### ISSUE THREAD

**Issue by LukasMa** [firehol/netdata#1087]
For me the postfix plugin isn't working properly. Error log shows this message:
`16-10-08 16:57:37: python.d ERROR: postfix_local Executing command postqueue -p resulted in error: [Errno 2] No such file or directory`
I also tried the full path to the postqueue executable but no luck. Any way to debug this properly?

**Comment by ktsaou** [comment]
hm... netdata tries to run it as user `netdata`. Check that this user can do it:

become user netdata with this:

```sh
su -s /bin/bash netdata
```

Can you run ` /usr/sbin/postqueue -p`?

**Comment by LukasMa** [comment]
Yes that works: `Mail queue is empty`.

**Comment by ktsaou** [comment]
hm... I don't know what is happening.
@paulfantom any ideas?

**Comment by paulfantom** [comment]
Usually `postqueue` is not in `PATH` env variable. Error presented by @LukasMa says just that, no `postqueue` in `PATH` (it is thrown in line 758 of `base.py`). However after this error, `ExecutableService` should be checking if `self.command` is in `/sbin` or `/usr/sbin` (line 791 in `base.py`).

@LukasMa can you show output of `python.d.plugin 1 debug postfix` ?

**Comment by LukasMa** [comment]
I'm running CentOS 7 and `postqueue` is on the `PATH`.

`$ ./python.d.plugin 1 debug postfix`:
```
16-10-09 14:33:36:  INFO: Using python v2
16-10-09 14:33:36: python.d INFO: reading configuration file: /etc/netdata/python.d.conf
16-10-09 14:33:36: python.d INFO: MODULES_DIR='/usr/libexec/netdata/python.d/', CONFIG_DIR='/etc/netdata/', UPDATE_EVERY=1, ONLY_MODULES=['postfix']
16-10-09 14:33:36: python.d DEBUG: postfix: loading module configuration: '/etc/netdata/python.d/postfix.conf'
16-10-09 14:33:36: python.d DEBUG: postfix: reading configuration
16-10-09 14:33:36: python.d DEBUG: postfix/local: job added
16-10-09 14:33:36: python.d DEBUG: all job objects [<Service(local, initial daemon)>]
16-10-09 14:33:36: python.d ERROR: postfix_local Executing command postqueue -p resulted in error: [Errno 2] No such file or directory
16-10-09 14:33:36: python.d DEBUG: postfix_local check succeeded
16-10-09 14:33:36: python.d DEBUG: overridden job names: []
16-10-09 14:33:36: python.d DEBUG: all remaining job objects: [<Service(, initial daemon)>]
CHART postfix_local.qemails '' 'Postfix Queue Emails' emails queue postfix.qemails line 60000 10
DIMENSION emails emails absolute 1 1
CHART postfix_local.qsize '' 'Postfix Queue Emails Size' 'emails size in KB' queue postfix.qsize area 60001 10
DIMENSION size size absolute 1 1

CHART netdata.plugin_pythond_postfix_local '' 'Execution time for postfix_local plugin' 'milliseconds / run' python.d netdata.plugin_python area 145000 1
DIMENSION run_time 'run time' absolute 1 1

16-10-09 14:33:36: python.d DEBUG: created charts for postfix_local
16-10-09 14:33:36: python.d DEBUG: postfix_local postfix_local will be run in 135 ms
16-10-09 14:33:37: python.d DEBUG: postfix_local postfix_local ready to run, after 136 ms (update_every: 1000 ms, latency: 0 ms
BEGIN postfix_local.qemails 0
SET emails = 0
END
BEGIN postfix_local.qsize 0
SET size = 0
END

BEGIN netdata.plugin_pythond_postfix_local 0
SET run_time = 14
END
```

**Comment by paulfantom** [comment]
By looking at your output of `python.d.plugin 1 debug postfix`, the postfix module IS working as it should. It created two charts:
```
CHART postfix_local.qemails '' 'Postfix Queue Emails' emails queue postfix.qemails line 60000 10
DIMENSION emails emails absolute 1 1
CHART postfix_local.qsize '' 'Postfix Queue Emails Size' 'emails size in KB' queue postfix.qsize area 60001 10
DIMENSION size size absolute 1 1
```
and then it updates their dimensions:
```
BEGIN postfix_local.qemails 0
SET emails = 0
END
BEGIN postfix_local.qsize 0
SET size = 0
END
```

Does running the same command produce similar output? If not it may be some kind of permission problem. Try setting SELinux into permissive mode and rerun it.

**Comment by LukasMa** [comment]
Yes that's true. I also see the charts in the dashboard. But I don't see the actual queue size. While `postqueue` gives me valid output, the chart is always showing `0`.

**BUT GOOD NEWS:**
I solved it! How? Well, I'm not sure. I completly removed netdata and installed it from scratch. And it works.
But I might know what could have been an issue: I modified the uid and gid of the netdata user in my previous installation. Unfortunaly I can't explain why exactly that should break the postfix plugin.
